qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/CreateAffiliationView.groovy

```diff
@@ -50,22 +51,43 @@ class CreateAffiliationView extends VerticalLayout {
     }
 
     private void initLayout() {
-        this.organisationField = new TextField("Organisation Name")
-        organisationField.setPlaceholder("name of the organisation")
-        organisationField.setDescription("Please enter the name of the organisation e.g. Universität Tübingen.")
-
-        this.addressAdditionField = new TextField("Department")
-        addressAdditionField.setPlaceholder("optional department name")
+        this.organisationBox = new ComboBox<>("Organisation Name")
+        
+        // we don't need the whole affiliation object, just the unique organization names.
+        List<String> organisationNames = sharedViewModel.affiliations.stream().map(affi -> affi.organisation).distinct().collect(Collectors.toList())
+
+        organisationBox.setItems(organisationNames)
+        organisationBox.setTextInputAllowed(true)
+        
+        // Check if the caption for new item already exists in the list of item
+        // captions before approving it as a new item.
+        ComboBox.NewItemProvider<String> itemHandler = newItemCaption -> {
+            boolean newItem = organisationNames.stream().noneMatch(data -> data.equalsIgnoreCase(newItemCaption))
+            if (newItem) {
+                // Adds new option
+                organisationNames.add(newItemCaption)
+                organisationBox.setItems(organisationNames)
+                organisationBox.setSelectedItem(newItemCaption)
+            }
+            return Optional.ofNullable(newItemCaption)
+        };
+        organisationBox.setNewItemProvider(itemHandler)
```

Comment:
Why not use a standard `ListDataProvider` and a caption generator as in the other view?

Reply:
I used code from the Vaadin sampler :) I'll have a look at the other class